source/Ship.cpp

```diff
@@ -951,6 +954,9 @@ void Ship::Save(DataWriter &out) const
 				out.Write("final explode", it.first->Name(), it.second);
 		});
 		
+		if(formationPattern)
+			out.Write("formation", formationPattern->Name());
```

Comment:
Are you referring to the situation where a ship refers to some formation-pattern which is not defined in the datafiles?

I still want to add some tests for this, but my impression is that this already gets handled well because an empty formation-pattern will be created upon load.
The only thing that would currently be missing is an warning/error-message about an empty formation being present.

Reply:
I looked at this a bit further and added a check in GameData::CheckReferences for formations where the Name is empty (formations that are references, but not defined).

I also added a check in FormationPattern::Load() to check for formations that are defined multiple times. (The formation pattern could look quite weird if two patterns are loaded on top of eachother.)